contracts/vesting/cw20-vest/src/contract.rs

```diff
@@ -0,0 +1,1219 @@
+                msg: to_binary(&cw20_stake::msg::ExecuteMsg::Claim {})?,
+                funds: vec![],
+            };
+            let transfer_msg = cosmwasm_std::WasmMsg::Execute {
+                contract_addr: config.token_address.to_string(),
+                msg: to_binary(&cw20::Cw20ExecuteMsg::Transfer {
+                    recipient: info.sender.to_string(),
+                    amount: release,
+                })?,
+                funds: vec![],
+            };
+            Ok(Response::new()
+                .add_message(claim_msg)
+                .add_message(transfer_msg)
+                .add_attribute("action", "claim")
+                .add_attribute("from", info.sender)
+                .add_attribute("amount", release))
+        } else {
+            let transfer_msg = cosmwasm_std::WasmMsg::Execute {
+                contract_addr: config.token_address.to_string(),
+                msg: to_binary(&cw20::Cw20ExecuteMsg::Transfer {
+                    recipient: info.sender.to_string(),
+                    amount: release,
+                })?,
+                funds: vec![],
+            };
+            Ok(Response::new()
+                .add_message(transfer_msg)
+                .add_attribute("action", "claim")
+                .add_attribute("from", info.sender)
+                .add_attribute("amount", release))
+        }
+    }
+
+    pub fn update_config(
+        deps: DepsMut,
+        _env: Env,
+        info: MessageInfo,
+        new_owner: Option<String>,
+        new_manager: Option<String>
+    ) -> Result<Response, ContractError> {
+        let new_owner = new_owner
+            .map(|new_owner| deps.api.addr_validate(&new_owner))
+            .transpose()?;
+        let new_manager = new_manager
+            .map(|new_manager| deps.api.addr_validate(&new_manager))
+            .transpose()?;
+
+        let mut config: Config = CONFIG.load(deps.storage)?;
+        if Some(info.sender.clone()) != config.owner && Some(info.sender.clone()) != config.manager {
+            return Err(ContractError::Unauthorized {});
+        };
+        if Some(info.sender) != config.owner && new_owner != config.owner {
+            return Err(ContractError::OnlyOwnerCanChangeOwner {});
+        };
+
+        config.owner = new_owner;
+        config.manager = new_manager;
+
+        CONFIG.save(deps.storage, &config)?;
+        Ok(Response::new()
+            .add_attribute("action", "update_config")
+            .add_attribute(
+                "owner",
+                config
+                    .owner
+                    .map(|a| a.to_string())
+                    .unwrap_or_else(|| "None".to_string()),
+            )
+            .add_attribute(
+                "manager",
+                config
+                    .manager
+                    .map(|a| a.to_string())
+                    .unwrap_or_else(|| "None".to_string()),
+            ))
+    }
+
+}
+
+pub mod query {
```

Comment:
probably want to add a query here for the config and one to get the vesting schedule ofr an address.

Reply:
Adding a vesting schedule query endpoint. Config should be there already.